src/firebase_functions/https_fn.py

```diff
@@ -362,23 +363,21 @@ def _on_call_handler(func: _C2, request: Request) -> Response:
             raise HttpsError(FunctionsErrorCode.UNAUTHENTICATED,
                              "Unauthenticated")
 
-        # TODO support for `allowInvalidAppCheckToken`
-        if token_status.app == _util.OnCallTokenState.INVALID:
+        if enforce_app_check and token_status.app == _util.OnCallTokenState.INVALID:
```

Comment:
Is this missing the case - if `enforce_app_check` is true and app check token is missing, should return unauthenticated.